### ISSUE TITLE
store state

### ISSUE THREAD

**Issue by ozum** [davestewart/vuex-pathify-demos#3]
Hi,

Great demos, well prepared. Just a kindly reminder, it is recommended to initialize state with a function instead of plain object. In demos state are initialized as plain object.

```js
export default {
  state: {
    value: 123,
  }
}
```

```js
export default () => ({
  state: {
    value: 123,
  }
})
```

**Comment by davestewart** [comment]
Hi Ozum,

AFAIK, state as an object is just fine.

Where did you see this recommendation?

I know that state itself supports functions which is useful for things like resetting:

```js
function reset () {
  return {
    value: 0,
  }
}

export default {
  state,
  mutations: {
    reset (state) {
      Object.assign(state, reset())
    }
  }
}
```

**Comment by davestewart** [comment]
Hi Ozum,

AFAIK, state as an object is just fine.

Where did you see this recommendation?

I know that state itself supports functions which is useful for things like resetting:

```js
function getState () {
  return {
    value: 0,
  }
}

export default {
  state: getState,
  mutations: {
    reset (state) {
      Object.assign(state, getState())
    }
  }
}
```

**Comment by ozum** [comment]
Yes, object is fine. Using function is recommended for module reuse as descibed in official Vuex docs [here](https://vuex.vuejs.org/guide/modules.html#module-reuse)

> This is actually the exact same problem with data inside Vue components. So the solution is also the same - use a function for declaring module state

**Comment by davestewart** [comment]
Yes, exactly :)

> So the solution is also the same - use a function for declaring module state

I think your example used a function for the entire module?

Did you intend that?

Also...

> Great demos, well prepared.

Thanks!

**Comment by ozum** [comment]
I use functions for every state initialization, because I saw it so in Vuex examples and read above part in docs. However  I’m very new to Vue and not experienced as you are. My interpretations may not be solid.

About demos... I’m new to vue, but experienced developer and, wow, demos are really great. Documentation, comments in codes, preparation, presentation and being ready to consume online👍 right away. I also saw your proactive effort in Nuxt community to bring your demo (and library) compatible to Nuxt.
